src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const currentState = { ...state };
```

Comment:
The variable name 'currentState' is a bit misleading since it implies that it's a single state that will be mutated. Since we need to avoid mutation of the original state and work with clones instead, a better name would be 'stateClone' or 'nextState'.